src/lib.rs

```diff
@@ -89,43 +90,66 @@ pub unsafe extern "C" fn rav1e_config_default() -> *mut Config {
     Box::into_raw(c)
 }
 
+/// Set color properties of the stream.
+///
+/// Supported values are defined by the enum types
+/// RaMatrixCoefficients, RaColorPrimaries, and RaTransferCharacteristics
+/// respectively.
+///
+/// Return a negative value on error or 0.
 #[no_mangle]
 pub unsafe extern "C" fn rav1e_config_set_color_description(cfg: *mut Config,
                                                             matrix: MatrixCoefficients,
                                                             primaries: ColorPrimaries,
-                                                            transfer: TransferCharacteristics)
-{
+                                                            transfer: TransferCharacteristics
+) -> c_int {
     (*cfg).cfg.enc.color_description = Some(rav1e::ColorDescription {
         matrix_coefficients: matrix,
         color_primaries: primaries,
         transfer_characteristics: transfer,
     });
+
```

Comment:
I guess you plan to do validation in the future.

Reply:
sorry what kind of validation? whether input could be represented in the related enum?